source/default/_patterns/01-atoms/button/index.js

```diff
@@ -7,6 +7,7 @@ import 'bootstrap/js/dist/button';
 
 // Custom
 import 'protons';
+import ParticleElement from '../../../_types/particle-element.ts';
```

Comment:
If we are going to go down the path of using the class, I think it would be nice to add a modules resolver in the Webpack config to point to `types` as a resolvable module directory. Then at any level of component depth we could do `import ParticleElement from 'types/particle-element'. We also can eliminate the `.ts` since it was added to the resolvable file extensions by webpack.

Here is a link to the webpack modules resolvers as well. https://webpack.js.org/configuration/resolve/#resolve-modules

Reply:
Perfect, will do! Thanks for the link too